src/drawing-tool/components/runtime.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useRef, useEffect } from "react";
+import DrawingTool from "drawing-tool";
+import 'drawing-tool/dist/drawing-tool.css';
+import { IAuthoredState, IInteractiveState } from "./app";
+import css from "./runtime.scss";
+import predefinedStampCollections from "./stamp-collections";
+import { renderHTML } from "../../shared/utilities/render-html";
+
+const kToolbarWidth = 40;
+const kToolbarHeight = 600;
+
+interface IProps {
+  authoredState: IAuthoredState;
+  interactiveState?: IInteractiveState | null;
+  setInteractiveState?: (updateFunc: (prevState: IInteractiveState | null) => IInteractiveState) => void;
+  report?: boolean;
+}
+
+interface StampCollections {
+  [collectionName: string]: string[];
+}
+
+interface DrawingToolOpts {
+  width: number;
+  height: number;
+  stamps?: StampCollections;
+}
+
+export const Runtime: React.FC<IProps> = ({ authoredState, interactiveState, setInteractiveState, report }) => {
+
+  const readOnly = !!(report || (authoredState.required && interactiveState?.submitted));
+
+  // equivalent to componentDidLoad
+  useEffect(() => {
+    const windowWidth = window.innerWidth;
+
+    const stampCollections: StampCollections = {};
+    authoredState.stampCollections?.forEach(collection => {
+      const baseName = collection.name || collection.collection.charAt(0).toUpperCase() + collection.collection.slice(1);
+      let name = baseName;
+      let i = 0;
+      while (!!stampCollections[name]) {
+        name = `${baseName} ${++i}`;
+      }
+      let stamps: string[];
+      if (collection.collection === "custom") {
+        stamps = collection.stamps || [];
+        const urlRegex = /^(https:|http:)\/\/\S+/;
+        stamps = stamps.map(url => url.replace(/ /g,'')).filter(url => url.match(urlRegex));
+      } else {
+        stamps = predefinedStampCollections[collection.collection];
+      }
+      if (stamps && stamps.length) {
+        stampCollections[name] = stamps;
+      }
+    });
+
+    const drawingToolOpts: DrawingToolOpts = {
+      width: windowWidth - kToolbarWidth - 10,
+      height: kToolbarHeight
```

Comment:
Interactive can be resized by LARA (eg. in the responsive layout). I guess you might need to use https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver to monitor changes to iframe dimensions (it's already used in this repo). Not sure if DrawingTool will like dynamic resizing. I guess it might be a new feature there.